### PR TITLE
Working PR for Treasury Dashboard (WIP)

### PR DESCRIPTION
**Pull request by jackiep00**
Creating a placeholder PR to track issues

### CODE REVIEW

File: treasury/dune-queries/treasury_balance_over_time_usd.sql [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    AND TYPE = 'create'
+)
+
+, days AS (
+    SELECT 
+        generate_series(MIN(day), date_trunc('day', NOW()), '1 day') AS day -- Generate all days since the first contract
+    FROM creation_days
+)
+
+, prices as (
```
Comment (josephd8) [inline treasury/dune-queries/treasury_balance_over_time_usd.sql:49]: So the `prices.usd` table will not have all prices going back to the start of the treasury wallet. You can see this in the results of this query, there are a lot of rows where the product and price are NULL. We will have to use manual price feeds where the `prices.usd` table does not contain data. I have already made queries for daily price feeds that combine a manual price feed and the `prices.usd` table:

INDEX - https://duneanalytics.com/queries/34732
DPI - https://duneanalytics.com/queries/30916
MVI - https://duneanalytics.com/queries/42184
ETH2x-FLI - WIP (will add shortly)

One idea would be to create one query that combines all Coop products price feeds. We can then use that as the prices CTE.

Let me know if this is making sense.

Comment (jackiep00) [reply treasury/dune-queries/treasury_balance_over_time_usd.sql:49]: i took a crack at creating a generalized index token price feed. it's in this PR and in this query here: https://duneanalytics.com/queries/46532

will resolve this comment if we QA it and determine it fit for service

File: treasury/dune-queries/generalized_price_feed.sql [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    SELECT
+        date_trunc('day', minute) AS dt
+        , symbol
+        , AVG(price) AS price
+    FROM prices.usd
+    WHERE symbol in ('INDEX', 'DPI', 'MVI', 'ETH2x-FLI')
+    GROUP BY 1,2
+)
+    
+, uni_swaps AS (
```
[the remark this answers is not in the document]
Comment (josephd8) [reply treasury/dune-queries/generalized_price_feed.sql:23]: @jackiep00

File: treasury/dune-queries/generalized_price_feed.sql [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    LEFT JOIN uni_swaps s ON h."hour" = s.hour 
+    LEFT JOIN uni_a1_prcs a ON h."hour" = a."hour"
+    GROUP BY 1,2
+
+) 
+, uni_feed AS (
+    SELECT
+        hour
+        , symbol
+        , (ARRAY_REMOVE(ARRAY_AGG(usd_price) OVER (ORDER BY hour), NULL))[COUNT(usd_price) OVER (ORDER BY hour)] AS usd_price
```
[the remark this answers is not in the document]
Comment (josephd8) [reply treasury/dune-queries/generalized_price_feed.sql:79]: @jackiep00

File: treasury/dune-queries/treasury_balance_over_time_usd.sql (hunk 1 above)
Comment (josephd8) [reply treasury/dune-queries/treasury_balance_over_time_usd.sql:50]: @jackiep00

File: treasury/dune-queries/generalized_price_feed.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (jackiep00) [reply treasury/dune-queries/generalized_price_feed.sql:79]: done!

File: treasury/dune-queries/generalized_price_feed.sql [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+)
+
+, swap_a1_prcs AS (
+
+    SELECT 
+        avg(price) a1_prc
+        , date_trunc('hour', minute) AS hour
+    FROM prices.usd
+    WHERE minute >= '2020-09-10'
+        AND contract_address ='\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' --weth as base asset
```
Comment (josephd8) [inline treasury/dune-queries/generalized_price_feed.sql:64]: Need to add wBTC as the base asset for the BTC2x swaps (not weth).

**Comment by jackiep00** [comment]
@josephd8 i checked with matt on the current status - seems like he's happy with it, and this basically has the components to continue iterating to deliver what he needs. I'm comfortable checking this in and call it this like v0.1 and starting a new PR for v0.2 for the next level granularity

**Comment by jackiep00** [comment]
that is, after I address the wBTC base asset comment above

**Comment by jackiep00** [comment]
Done - i also created a separate folder called "helper CTEs" so it's clear that you can use those components in other complex queries.

**Pull request merged**